### PR TITLE
support custom labeling

### PR DESCRIPTION
**Pull request by SharonGrossman**
- add support of labeling
- .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,15 +100,15 @@ Invoke a function that returns a global context to use in all monitor invocation
...
+|:-----------------------------------------------------:|-----------------------------------------------------------------------------|
+|                  `context?: boolean`                  | add context that will be logged in all method's logs                        | 
+|                 `logResult?: boolean`                 | log the method's result                                                     | 
+|             `logExecutionStart?: boolean`             | log the start of the method's execution `method.start`                      |
+|            `parseResult?: (e: any) => any`            | transform the method's result that will be returned                         |
+|            `parseError?: (e: any) => any`             | if the method errored, transform the error that will be thrown              |
+|             `errorLogLevel?: pino.Level`              | if the method errored, which level should the message be, default - `error` |
+  | `labeling?: {name: string, contextKeyPath: string}[]` | add custom labeled counters using context keys and values
```
Comment (noamokman) [inline README.md:111]: this is not aligned and missing the last `|`

File: README.md [hunk 2]
```diff
@@ -50,6 +50,9 @@ const errored = (id: string) => await monitor('query', async () => db.query(id),
 
 // Log Execution Start
 const executionStart = (id: string) => await monitor('query', async () => db.query(id), { logExecutionStart: true });
+
+// Custom labeling
+const customLabels = (id: string) => await monitor('query', async () => db.query(id), { labeling: [{ name: 'labelName', contextKeyPath: 'key.path.id' }] });
```
Comment (noamokman) [inline README.md:55]: this make no sense if you don't use a context, add it to the example

File: src/monitor.ts [hunk 3]
```diff
@@ -101,7 +104,7 @@ const innerMonitor = <Callable>({ scope: monitorScope, method: monitorMethod, ca
         throw error;
       }) as any as Callable;
   } catch (error) {
-    counter.inc({ method, result: 'error' });
+    counter.inc({ method, result: 'error', ...labelValues });
```
Comment (noamokman) [inline src/monitor.ts:107]: we should spread it before the rest so someone won't override these by mistake

File: src/types.ts [hunk 4]
```diff
@@ -22,6 +22,12 @@ export interface MonitorOptionsBase {
 
 export interface MonitorOptions<T> extends MonitorOptionsBase {
   parseResult?: (r: Awaited<T>) => any;
+  labeling?: ContextLabeling[];
+}
+
+export interface ContextLabeling {
+  name: string;
+  path: string;
```
Comment (omerlh) [inline src/types.ts:30]: Why not just value? Why take it from context?

Comment (noamokman) [reply src/types.ts:30]: oh yeah, it can just be an object for labels

Comment (SharonGrossman) [reply src/types.ts]: Fixed, just passing name and value right now, simpler

**Review by omerlh — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -50,6 +50,9 @@ const errored = (id: string) => await monitor('query', async () => db.query(id),
 
 // Log Execution Start
 const executionStart = (id: string) => await monitor('query', async () => db.query(id), { logExecutionStart: true });
+
+// Custom labeling
+const customLabels = (id: string) => await monitor('query', async () => db.query(id), {context: {key: 'myId' }, labeling: [{ name: 'labelName', value: 'myId' }] });
```
Comment (noamokman) [inline README.md:55]: We can just change this to a dictionary, and then the code will be even simpler

**Review by noamokman — APPROVED** [review]

**Pull request merged**